qt/python/mantidqt/widgets/algorithmselector/test/observer_test.py

```diff
@@ -0,0 +1,79 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantid workbench.
+#
+#
+from __future__ import absolute_import
+
+import sys
+import unittest
+
+from mantid.api import AlgorithmFactory, PythonAlgorithm
+from mantidqt.widgets.algorithmselector.widget import AlgorithmSelectorWidget
+from mantidqt.widgets.algorithmselector.algorithm_factory_observer import AlgorithmSelectorFactoryObserver
+
+if sys.version_info.major >= 3:
```

Comment:
A shim was added recently to mantid so that you could do `from mantid.py3compat import mock` instead of needing to do this check everywhere. Could we use this here and in the other test?